src/main/java/com/paneedah/mwc/items/guns/MP40Factory.java

```diff
@@ -3,1517 +3,159 @@
 import com.paneedah.mwc.MWC;
 import com.paneedah.mwc.models.weapons.*;
 import com.paneedah.mwc.proxies.CommonProxy;
-import com.paneedah.mwc.weapons.Attachments;
-import com.paneedah.mwc.weapons.AuxiliaryAttachments;
-import com.paneedah.mwc.weapons.Magazines;
-import com.paneedah.weaponlib.RenderableState;
-import com.paneedah.weaponlib.Weapon;
-import com.paneedah.weaponlib.WeaponRenderer;
-import com.paneedah.weaponlib.animation.Transition;
+import com.paneedah.mwc.weapons.*;
+import com.paneedah.weaponlib.*;
+import com.paneedah.mwc.rendering.Transform;
+import com.paneedah.weaponlib.compatibility.RecoilParam;
 import com.paneedah.weaponlib.config.BalancePackManager.GunConfigurationGroup;
 import com.paneedah.weaponlib.render.shells.ShellParticleSimulator.Shell.Type;
 import net.minecraft.item.Item;
+import net.minecraft.util.math.Vec3d;
 import org.lwjgl.opengl.GL11;
 
-import java.util.Arrays;
-
 public class MP40Factory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("mp40")
+                .withFireRate(0.6f)
```

Comment:
No balance changes until 0.2.
```suggestion
                .withFireRate(0.55f)
```